### PR TITLE
feat: Add ability to use multiple scopes

### PR DESCRIPTION
**Pull request by amromusharaf**
This PR adds the oft-requested ability to use multiple scopes, and includes tests. For example:

feat(foo,bar): Add cool new feature

### CODE REVIEW

File: src/validatePrTitle.js [hunk 1]
```diff
@@ -54,11 +58,13 @@ module.exports = async function validatePrTitle(
     );
   }
 
-  if (scopes && result.scope && !scopes.includes(result.scope)) {
+  const givenScopes = result.scope ? result.scope.split(',') : undefined;
```
Comment (amannn) [inline src/validatePrTitle.js:61]: Maybe trim the individual parts to allow for `fix(core, ui):`?

```suggestion
  const givenScopes = result.scope ? result.scope.split(',').map(scope => scope.trim()) : undefined;
```

File: src/validatePrTitle.test.js [hunk 2]
```diff
@@ -55,11 +55,25 @@ describe('defined scopes', () => {
...
+    await validatePrTitle('fix(core,e2e): Bar', {
+      scopes: ['core', 'e2e', 'web']
+    });
+  });
+
+  it('throws when an unknown scope is detected within multiple scopes', async () => {
+    await expect(
+      validatePrTitle('fix(core,e2e,foo,bar): Bar', {scopes: ['foo', 'core']})
+    ).rejects.toThrow(
+      'Unknown scope(s) "e2e,bar" found in pull request title "fix(core,e2e,foo,bar): Bar". Use one of the available scopes: foo, core.'
```
Comment (amannn) [inline src/validatePrTitle.test.js:68]: Nice that you only list the unknown scopes 👍.

File: src/validatePrTitle.js [hunk 3]
```diff
@@ -54,11 +58,13 @@ module.exports = async function validatePrTitle(
...
+  const unknownScopes = givenScopes ? givenScopes.filter(isUnknownScope) : [];
+  if (scopes && unknownScopes.length > 0) {
     throw new Error(
-      `Unknown scope "${
-        result.scope
-      }" found in pull request title "${prTitle}". Use one of the available scopes: ${scopes.join(
+      `Unknown scope(s) "${unknownScopes.join(
```
Comment (amannn) [inline src/validatePrTitle.js:65]: Let's handle the plural for slightly better error messages:

```suggestion
      `Unknown ${unknownScopes.length > 1 ? 'scopes' :'scope'} "${unknownScopes.join(
```

**Review by amannn — CHANGES_REQUESTED** [review]
Hey @amromusharaf,

thank you for this excellent pull request! 💯

I only left some minor comments, maybe give them your consideration.

Do you know if other tools like `semantic-release` support multiple scopes as well? The combination with that tool was at least my initial motivation to build this action. If yes, maybe it's worth mentioning this feature in the README with the validation examples.

Otherwise I'd silently support it the way you've implemented it right now. Otherwise a user might get the impression that this syntax is generally supported.

Thanks!

**Comment by amromusharaf** [comment]
Hey @amannn -- thanks for your quick response! The changes you've requested sound great; I've pushed them, and I appreciate your thoughtful review :) 

As to the question re: `semantic-release`, I'm unsure whether or not they support multiple scopes, though I'd be a little surprised if they do, as I'm not sure this is part of the official spec just yet. That being said, I noticed it's been requested in a few places, which is why I ultimately ended up opening this PR:

https://www.conventionalcommits.org/en/v1.0.0/#specification
https://github.com/conventional-commits/conventionalcommits.org/issues/302
https://github.com/conventional-changelog/commitlint/issues/75
https://github.com/conventional-changelog/conventional-changelog/issues/232

I think, as you say, it may be best to silently support it for the time being?

Thanks again for your time, by the way -- it's been a pleasure contributing to your codebase :)

File: src/validatePrTitle.js (hunk 3 above)
Comment (amromusharaf) [reply src/validatePrTitle.js]: 👍

File: src/validatePrTitle.test.js (hunk 2 above)
Comment (amromusharaf) [reply src/validatePrTitle.test.js]: 😄

File: src/validatePrTitle.js (hunk 1 above)
Comment (amromusharaf) [reply src/validatePrTitle.js]: 💯

**Comment by amannn** [comment]
> I noticed it's been requested in a few places

True, thank you for the links. It seems like commit lint also supports the same syntax already: https://github.com/conventional-changelog/commitlint/pull/901.

> I think, as you say, it may be best to silently support it for the time being?

Yep, sounds fair!

> it's been a pleasure contributing to your codebase :)

Great to hear! Thank you for your contribution 👏

File: src/validatePrTitle.test.js [hunk 4]
```diff
@@ -55,11 +55,25 @@ describe('defined scopes', () => {
...
 
+  it('allows multiple matching scopes', async () => {
+    await validatePrTitle('fix(core,e2e): Bar', {
+      scopes: ['core', 'e2e', 'web']
+    });
+  });
+
+  it('throws when an unknown scope is detected within multiple scopes', async () => {
+    await expect(
+      validatePrTitle('fix(core,e2e,foo,bar): Bar', {scopes: ['foo', 'core']})
```
Comment (webstoreportal) [inline src/validatePrTitle.test.js:66]: any issue in convention of comma separator symbol `,` with—or without—a trailing space?

**Review by webstoreportal — COMMENTED** [review]
any issue in convention of comma separator symbol `,` with—or without—a trailing space?

File: src/validatePrTitle.test.js (hunk 4 above)
Comment (amannn) [reply src/validatePrTitle.test.js:66]: Not sure about the official convention, but I think we should support spaces in this action:

https://github.com/amannn/action-semantic-pull-request/blob/1829a7c0a510f3f783b5463c8c05923802e7de3a/src/validatePrTitle.js#L62

**Pull request merged**